include/bl_storage.h

```diff
@@ -106,6 +165,169 @@ uint16_t get_monotonic_counter(void);
  */
 int set_monotonic_counter(uint16_t new_counter);
 
+/**
+ * @brief The PSA life cycle states a device can be in.
+ *
+ * The LCS can be only transitioned in the order they are defined here.
+ */
+enum lcs {
+	BL_STORAGE_LCS_UNKNOWN = 0,
+	BL_STORAGE_LCS_ASSEMBLY = 1,
+	BL_STORAGE_LCS_PROVISIONING = 2,
+	BL_STORAGE_LCS_SECURED = 3,
+	BL_STORAGE_LCS_DECOMMISSIONED = 4,
+};
+
+/**
+ * Copies @p src into @p dst. Reads from @p src are done 32 bits at a
+ * time. Writes to @p dst are done a byte at a time.
+ *
+ * This ensures UICR->OTP's word-aligned and word-sized read
+ * requirement is met and also allows the destination buffer to be
+ * free of this requirement.
+ *
+ * @param[out] dst destination buffer.
+ * @param[in] src source buffer.
+ * @param[in] size number of *bytes* in src to copy into dst.
+ */
+NRFX_STATIC_INLINE void otp_copy32(uint8_t *restrict dst, uint32_t volatile * restrict src,
+				   size_t size)
+{
+	for (int i = 0; i < size / 4; i++) {
+		uint32_t val = src[i];
+
+		for (int j = 0; j < 4; j++) {
+			dst[i * 4 + j] = (val >> 8 * j) & 0xFF;
+		}
+	}
```

Comment:
This doesn't currently support `size`s that aren't word-aligned (It will currently copy 8 bytes if size is 10). It also doesn't support `src` being unaligned.

IMO it should either support these, or clearly state the limitiation in the docs, and possibly check `size` and/or `src` at run time.

The following adds support for unaligned `size`s. It's not tested by me :P, so do a check if you use it. `ROUND_UP` and `MIN` are found in `#include <zephyr/sys/util.h>`
```suggestion
	size_t size_copy = size;
	for (int i = 0; i < ROUND_UP(size, 4) / 4; i++) {
		uint32_t val = src[i];

		for (int j = 0; j < MIN(4, size_copy); j++) {
			dst[i * 4 + j] = (val >> 8 * j) & 0xFF;
		}
		size_copy -= 4;
	}
```

Reply:
Do you feel that these docs do not suffice?

https://github.com/nrfconnect/sdk-nrf/pull/9033/files/848acd581c510144ac24f7de8250e9a5cdf5b8ba#diff-330779901eea51e2eb8f11a270058dde977fd0dd60808342e713bd4b1d3a5e78R182-R187

Reply:
Agreed